aleph0/src/createActions.ts

```diff
@@ -25,20 +42,45 @@ export const createActions = (
         properties: {},
       },
     },
-    readSnippetFile: {
-      function: async (args: { filePath: string }) => {
-        return { fileContents: readFile(args.filePath) };
+    expandSnippet: {
+      function: async (args: { id: string }) => {
+        const { file, references } = files[args.id];
+
+        const directory = process.cwd();
+
+        const filePath = `${directory}/${file}`;
+        const fileContents = readFile(filePath);
+        const referenceContents = references?.map((reference) => {
+          return {
+            name: reference,
+            contents: readFile(`${directory}/${reference}`),
+          };
+        });
+
+        const contents = `## File:
+${fileContents}
+
+## References:
+
+${referenceContents?.map(
+  (reference) => `### ${reference.name}
+${reference.contents}`
+)}
+`;
+
+        return contents;
       },
-      name: "readSnippetFile",
-      description: "Returns the contents of a snippet file.",
+      name: "expandSnippet",
+      description: "Returns the contents and references of a snippet file.",
```

Comment:
Maybe give the ai more explaining of what a reference is and how it helps? (also maybe that it has to call it before using a snippet otherwise it will lack context).

Reply:
Not sure how they help. It's just additional info for the AI to use.

Reply:
We can wait to see if it makes problems. But from my understanding giving it that extra information usually helps. So it will have an easier time inferring when to call the reference into context.

Reply:
It will insert the reference files too. Not sure what they're supposed to be. You can add info to it if you want. The references themselves may have that info for it already.